tests/cleanup_resources.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) Microsoft Corporation.
+# Adapted from original code by Jeremi Piotrowski <jpiotrowski@microsoft.com>.
+from datetime import datetime, timedelta, timezone
+import os
+
+from azure.identity import AzureCliCredential
+from azure.mgmt.resource import ResourceManagementClient
+
+# Current date and time in UTC.
+utc_now = datetime.now(timezone.utc)
+# Cleanup time delta passed in the environment, i.e. how much time to
+# wait before automatically cleaning up a resource.
+cleanup_after = timedelta(hours=int(os.environ['CLEANUP_AFTER_HOURS']))
+# Time considered as the cutoff to clean up a resource: if it was
+# created before this time, it will be deleted.
+cleanup_cutoff_time = utc_now - cleanup_after
+
+print(f"Current time: {utc_now}")
+print(f"Cleanup time delta: {cleanup_after}")
+print(f"Will clean up resources created before {cleanup_cutoff_time}")
+
+credential = AzureCliCredential()
+subscription_id = os.environ['AZ_SUBSCRIPTION_ID']
+client = ResourceManagementClient(credential, subscription_id)
+resources = client.resources.list(expand='createdTime')
+
+print("Processsing resources...")
+num_deleted = 0
+
+for resource in resources:
+    # Ignore resources that aren't AKS clusters.
+    if resource.type != 'Microsoft.ContainerService/managedClusters':
+        continue
+
+    # Ignore resources created after the cutoff time (i.e. less than
+    # `cleanup_after` time ago).
+    if resource.created_time > cleanup_cutoff_time:
+        print(f"{resource.name}: created at {resource.created_time}, after delta cutoff, ignored")
+        continue
+
+    print(f"{resource.name}: created at {resource.created_time}, before delta cutoff, deleting...")
+
+    # XXX DANGER ZONE: Delete the resource.
+    client.resources.begin_delete_by_id(resource.id)
```

Comment:
We create a unique resource group per cluster - should we be cleaning that up as well?

Reply:
Good point, just added code to do this as long as it's the only resource in the resource group.